layout-normalizer/src/gpos.rs

```diff
@@ -228,6 +239,22 @@ impl ResolvedValue {
     fn is_zero(&self) -> bool {
         self.default == 0 && self.device_or_deltas.is_none()
     }
+
+    fn add_in_place(&mut self, other: &ResolvedValue) {
+        self.default += other.default;
+        //CODE REVIEW:
+        // - if we have device tables we're just dropping the second, which
+        // is fine for our purposes? And if we have mixed deltas & devices than
+        // god help us
+        if let (Some(DeviceOrDeltas::Deltas(d1)), Some(DeviceOrDeltas::Deltas(d2))) = (
+            self.device_or_deltas.as_mut(),
+            other.device_or_deltas.as_ref(),
+        ) {
+            // I think this is expected/required?
+            assert_eq!(d1.len(), d2.len());
```

Comment:
well, in theory the number of deltas might be different in different lookups.. We can't merge them if they aren't.

Reply:
hmm. In my code we are always using the same item variation store and the same set of master locations when resolving the deltas. Are you thinking about sparseness? In that case I think we're fine, and should be inserting zeros, although maybe I'm misunderstanding?

Reply:
yes i'm thinking about sparse masters, even if right now fontc doesn't emit them for the generated GPOS features (https://github.com/googlefonts/fontc/issues/407) it doesn't mean pos rules have to always be "dense" and contain all the same number of masters.
It's not as simple as inserting zeros, I'm afraid. It is conceivable that an item variation store has multiple vardata tables each specifying different number of regions and then various pos rules referencing one or the other.

Reply:
hmm okay I'm worried.


This is how, in `layout-normalizer`, we compute the deltas (they aren't actually deltas; they're absolute value at each master location?) https://github.com/googlefonts/fontc/blob/dab7473fe4ae791deaf5fb934b1fb78493829de5/layout-normalizer/src/variations.rs#L50

For a given set of master locations, this will always return the same number of values. Is my logic wrong somewhere?

Reply:
Yes, one delta for each master location, but each given coordinate could potentially be defined along a different number of master locations

Reply:
I'm sorry but I'm still confused. We only use a single set of master locations, which we retrieve from the IVS itself; then for each location we call `ItemVariationStore::compute_delta`, which adds up the deltas for all the regions that are active for that coordinate; so there are potentially a different number of deltas but there are always the same number of total master locations?

Reply:
> We only use a single set of master locations, which we retrieve from the IVS itself

then we are wrong, because the IVS has a global VarRegionList and then each VariationData table can reference different sets of regions, and the various variable things that can have a VarIndex are free to reference one or the other VarData, effectively selecting only some from the IVS global list of regions

Reply:
maybe the problem is just my naming? As in this is called 'deltas' but it's really the resolved value for each defined master/location.

Reply:
no, it's not just naming, it's wrong to assume a single set of master locations (aka. region peaks) for the entire IVS like our DeltaComputer is doing. There are potentially many sets of master locations, one per VariationData subtable